engine/src/test/java/org/kigalisim/validate/LiveTests.java

```diff
@@ -340,7 +340,174 @@ public void testBasicUnitsConvert() throws IOException {
+      ParsedPolicy policy = program.getPolicy(policyName);
+      assertNotNull(policy, "Policy " + policyName + " should exist");
+      executePolicy(policy, machine);
+    }
+
+    // Move to year 10 and check results (like the JavaScript test)
+    for (int year = 2; year <= 10; year++) {
+      engine.incrementYear();
+    }
+
+    List<EngineResult> results = engine.getResults();
+
+    // Check Sub A in year 10 - should have 0 GHG consumption due to replacement
+    EngineResult resultSubA10 = findResult(results, "Test", "Sub A", 10);
+    assertNotNull(resultSubA10, "Should have result for Test/Sub A in year 10");
+
+    // Check Sub B in year 10 - should have GHG consumption due to replacement
+    EngineResult resultSubB10 = findResult(results, "Test", "Sub B", 10);
+    assertNotNull(resultSubB10, "Should have result for Test/Sub B in year 10");
+
+    // Verify that Sub B has GHG consumption in year 10 due to replacement
+    System.out.println("RETIRE ONLY - Sub A Year 10 GHG: " + resultSubA10.getGhgConsumption().getValue() + " " + resultSubA10.getGhgConsumption().getUnits());
+    System.out.println("RETIRE ONLY - Sub B Year 10 GHG: " + resultSubB10.getGhgConsumption().getValue() + " " + resultSubB10.getGhgConsumption().getUnits());
+    assertTrue(resultSubB10.getGhgConsumption().getValue().doubleValue() > 0,
+        "Sub B should have non-zero GHG consumption in year 10 due to replacement (retire only)");
+  }
+
+  /**
+   * Test basic_replace with recharge only - to isolate the issue.
+   */
+  @Test
+  public void testBasicReplaceRechargeOnly() throws IOException {
+    // Load and parse the QTA file
+    String qtaPath = "/tmp/test_basic_replace_recharge_only.qta";
+    ParsedProgram program = KigaliSimFacade.parseAndInterpret(qtaPath);
+    assertNotNull(program, "Program should not be null");
+
+    // Get the scenario
+    String scenarioName = "Sim";
+    ParsedScenario scenario = program.getScenario(scenarioName);
+    assertNotNull(scenario, "Scenario should exist");
+
+    // Get start and end years from scenario
+    int startYear = scenario.getStartYear();
+    int endYear = scenario.getEndYear();
+
+    // Create engine and machine
+    Engine engine = new SingleThreadEngine(startYear, endYear);
+    PushDownMachine machine = new SingleThreadPushDownMachine(engine);
+
+    // Execute the default policy
+    ParsedPolicy defaultPolicy = program.getPolicy("default");
+    assertNotNull(defaultPolicy, "Default policy should exist");
+    executePolicy(defaultPolicy, machine);
+
+    // Execute other policies in the scenario
+    for (String policyName : scenario.getPolicies()) {
+      ParsedPolicy policy = program.getPolicy(policyName);
+      assertNotNull(policy, "Policy " + policyName + " should exist");
+      executePolicy(policy, machine);
+    }
+
+    // Move to year 10 and check results (like the JavaScript test)
+    for (int year = 2; year <= 10; year++) {
+      engine.incrementYear();
```

Comment:
This is not enough to actually run the simulation from year to year. We should transition `LiveTests` as a whole to use `KigaliSimFacade`